core/src/main/java/org/mule/runtime/core/processor/strategy/LegacySynchronousProcessingStrategyFactory.java

```diff
@@ -6,39 +6,20 @@
  */
 package org.mule.runtime.core.processor.strategy;
 
-import org.mule.runtime.core.api.Event;
 import org.mule.runtime.core.api.MuleContext;
 import org.mule.runtime.core.api.processor.strategy.ProcessingStrategy;
 import org.mule.runtime.core.api.processor.strategy.ProcessingStrategyFactory;
 
-import java.util.function.Consumer;
-
 /**
- * This processing strategy processes all message processors in the calling thread.
+ * Processing strategy used to force legacy 3.x synchronous behaviour that uses a blocking code path.
  */
 @Deprecated
 public class LegacySynchronousProcessingStrategyFactory implements ProcessingStrategyFactory {
 
-  static ProcessingStrategy LEGACY_SYNCHRONOUS_PROCESSING_STRATEGY_INSTANCE = new AbstractLegacyProcessingStrategy() {
-
-    @Override
-    public boolean isSynchronous() {
-      return true;
-    }
-
-    /*
-     * This processing strategy supports transactions so we override default check that fails on transactions.
-     */
-    @Override
-    protected Consumer<Event> createOnEventConsumer() {
-      return event -> {
-      };
-    }
-  };
-
   @Override
   public ProcessingStrategy create(MuleContext muleContext, String schedulersNamePrefix) {
-    return LEGACY_SYNCHRONOUS_PROCESSING_STRATEGY_INSTANCE;
+    return (flowConstruct, pipelineFunction) -> {
+      throw new IllegalStateException("Sink cannot be created for " + LegacySynchronousProcessingStrategyFactory.class.getName());
```

Comment:
In this case i think illegalState is more specific/correct, as this should never be called do it if conditions in AbstractPipeline/Flow